cgen/__init__.py

```diff
@@ -26,6 +26,18 @@
 
 import numpy
 from pytools import memoize_method, memoize
+import warnings
+
+from cgen.preprocessor import Generable, Block, Collection, Comment, Line
```

Comment:
These don't make sense in that submodule.

You could create an "infrastructure" submodule perhaps.

Reply:
I agree, those classes should not be part of the preprocessor source file, I will move Generable, ```Block```, ```Collection``` and ```Line``` to an **infrastructure.py** file. I guess the ```Comment``` class should be part of the **preprocessor.py** file (at least not the new **infrastructure.py**).